### PR TITLE
Progress based

### PR DESCRIPTION
**Pull request by kevers-google**
Adds support for progress-based animations.

### CODE REVIEW

File: src/proxy-animation.js [hunk 1]
```diff
@@ -498,6 +580,140 @@ function notifyReady(details) {
...
+          nativeUpdateTiming.apply(effect, [timing]);
+        }
+      }
+      details.normalizedTiming = timing;
+      return details.specifiedTiming;
+    }
+  };
+  const updateTimingHandler = {
+    apply: function(target, thisArg, argumentsList) {
+      // Additional validation that is specific to scroll timeilnes.
```
Comment (flackr) [inline src/proxy-animation.js:683]: nit: s/timeilnes/timelines

File: src/proxy-animation.js [hunk 2]
```diff
@@ -543,10 +768,20 @@ export class ProxyAnimation {
...
+    // Proxy the effect to support timing conversions for progress based
+    // animations.
+    if (!details.effect)
+      details.effect = createProxyEffect(details);
+
+    return details.effect;
   }
   set effect(newEffect) {
     proxyAnimations.get(this).animation.effect = newEffect;
+    details.proxyEffect = null;
```
Comment (flackr) [inline src/proxy-animation.js:784]: Did you mean details.effect = null ?

File: src/proxy-animation.js [hunk 3]
```diff
@@ -534,6 +751,14 @@ export class ProxyAnimation {
...
       pendingTask: null,
+      // Record the specified timing since it may be different than the timing
+      // actually used for the animation. When fetching the timing, this value
+      // will be returned, however, the native animation will use normalized
+      // values.
+      specifiedTiming: null,
+      // The normalized timing has the corrected timing with the intrinsic
+      // iteration duration resolved.
+      normalizedTiming: null,
       proxy: this
```
Comment (flackr) [inline src/proxy-animation.js:762]: Can we add effect to this list since we're using that to cache the proxy effect?

**Review by flackr — COMMENTED** [review]
This is awesome! Great work proxying the specified and computed timing 😄 

I think this looks good, just some minor concerns throughout.

File: src/proxy-animation.js [hunk 4]
```diff
@@ -43,11 +43,64 @@ function createAbortError() {
   return new DOMException("The user aborted a request", "AbortError");
 }
 
+function toCssNumberish(details, value) {
+  if (value === null)
+    return value;
+
+  if (typeof value !== 'number') {
+      throw new DOMException(
```
[the remark this answers is not in the document]
Comment (kevers-google) [reply src/proxy-animation.js:51]: toCssNumberish is called from numerous places such as set currentTime, which are developer exposed. We do need to guard against garbage input.

[the remark this answers is not in the document]
Comment (flackr) [reply src/proxy-animation.js:51]: Got it, makes sense!

File: src/proxy-animation.js (hunk 3 above)
Comment (kevers-google) [reply src/proxy-animation.js:770]: done

File: src/proxy-animation.js (hunk 2 above)
Comment (kevers-google) [reply src/proxy-animation.js]: done

File: src/proxy-animation.js (hunk 1 above)
Comment (kevers-google) [reply src/proxy-animation.js]: done

File: src/proxy-animation.js [hunk 5]
```diff
@@ -43,11 +43,64 @@ function createAbortError() {
   return new DOMException("The user aborted a request", "AbortError");
 }
 
+function toCssNumberish(details, value) {
```
[the remark this answers is not in the document]
Comment (kevers-google) [reply src/proxy-animation.js:50]: done

**Review by flackr — APPROVED** [review]

**Pull request merged**